solr/solr-ref-guide/modules/query-guide/pages/dense-vector-search.adoc

```diff
@@ -109,20 +109,20 @@ and the hyper-parameter of the HNSW algorithm make sure you set this configurati
 Here's how `DenseVectorField` can be configured with the advanced codec hyper-parameters:
 
 [source,xml]
-<fieldType name="knn_vector" class="solr.DenseVectorField" vectorDimension="4" similarityFunction="cosine" codecFormat="Lucene90HnswVectorsFormat" hnswMaxConnections="10" hnswBeamWidth="40"/>
+<fieldType name="knn_vector" class="solr.DenseVectorField" vectorDimension="4" similarityFunction="cosine" codecFormat="Lucene91HnswVectorsFormat" hnswMaxConnections="10" hnswBeamWidth="40"/>
 <field name="vector" type="knn_vector" indexed="true" stored="true"/>
 
 `codecFormat`::
 +
 [%autowidth,frame=none]
 |===
-|Optional |Default: `Lucene90HnswVectorsFormat`
+|Optional |Default: `Lucene91HnswVectorsFormat`
 |===
 +
 (advanced) Specifies the knn codec implementation to use
 +
 
-Accepted values: `Lucene90HnswVectorsFormat`.
+Accepted values: `Lucene91HnswVectorsFormat`.
```

Comment:
Should we be able to accept the Lucene90HnswVectorsFormat here as well? I'm not an expert on this, but from reading LUCENE-10054 it looks like this format is not entirely compatible with the old one... Should we give users the option to stay on the same format?

Reply:
The experimental Dense Vector Search under development in Lucene (and Solr consequentially) is expected to change a lot in this phase.
There have been many changes both in Lucene 9.1.0 and in the upcoming 9.2.0.

Some of these changes are backed by a new codec effectively, so it's a delicate balance.

When features are marked as "Experimental" is it acceptable to make it simple and just require people to re-index(of course only if they use a vector-based field)?

That codec param is going away soon anyway(we'll submit a PR after this for that), as it's preferable to give the user the ability to just choose the algorithm (currently only HNSW supported) rather than the internal codec.

@msokolov , @mocobeta  what do you think?
Thanks @madrob  for raising this, it's definitely something we need to discuss!

Reply:
Hi, thanks for working on it.
I was not involved in the recent format changes in the vector values, but I don't think Lucene 9.1 can read `Lucene90HnswVectorsFormat` as @alessandrobenedetti explained - I haven't tried yet, I'll check it this week and report my observation here.
I noticed we don't have a MIGRATE entry in Lucene for the file format change. I think it'd be better to have some notes on reindexing to avoid unnecessary confusion.

Reply:
and there is going to be a 9.3 with still more changes, but we will maintain backcompat throughout the 9.x series. There's a statement to that effect here: https://cwiki.apache.org/confluence/display/lucene/BackwardsCompatibility

Reply:
@msokolov thanks, I didn't notice we have `lucene90` and `lucene91` in backward-codecs. So it should work if the applications have the backward codecs jar - I'll just check it.

Reply:
I manually checked that Lucene 9.1.0 reads the vector index created by Lucene 9.0 (with `lucene-backward-codecs` jar). Coming 9.2.0 reads the vectors written by 9.0 and 9.1 as well.
We keep all codecs in the same and the previous major version in backward-codecs, there is no need to care about the file format backward compatibility in an older but same major version from the users' perspective (Solr maintainers would properly add the dependency on backward-codecs module).

Reply:
Solr always adds lucene-backward-codec jars.

Reply:
And yes, like for other codecs, we should not lead the user to stick with specific formats - int´stead try to emphasize that the defaults are fine. Selecting different codec formats was always for special users only with a big red warning. Now the should go away from documentation or should be aligned to the other codec settings like BlockTerms.

Reply:
The problem of that setting is that the SchemaCodecFactory also has those hardcoded names. IMHO this should be changed in Solr, so SChemaCodecFactory should use the defaults from Lucene. Maybe every comonent should have a link to the Default (static final var). We have `Codec.getDefault()`, but maybe we need that for al format variants (PostingsFormat).

When Solr was part of Lucene, this was easier to maintain, but now it would be better to make Lucene not only expose the default codec, but also defaults for its components. E.g. `PostingsFormat.getDefault()`, `VectorsFormat.getDefault()`.

Reply:
I agree to remove the possibility to use custom Codec and use the Lucene default. However, I think it is better to open a specific Jira issue for that.
For this pull request, I reintroduced the possibility to configure Lucene90HnswVectorsFormat in addition to Lucene91HnswVectorsFormat and I updated the documentation.

Reply:
Thanks. I did also not understand the backwards compatibility problem. The codec is written to index file format, so when the backwards-codec JAR file is there (shipped with Solr), it can open old index. The Codec definition in schema is only used for new segments in index.
So thanks for removing the migration note.